### PR TITLE
Enhancements for kya (Excluding Package.lock)

### PR DESCRIPTION
**Pull request by BenjaminSsempala**
**_WHAT DOES THIS PR DO?_**
- Enhancements for kya

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []()
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

File: src/device-registry/models/KnowYourAirLesson.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ knowYourAirLessonSchema.statics = {
       };
     }
   },
-  async list({ skip = 0, limit = 1000, filter = {} } = {}) {
+  async list({ skip = 0, limit = 1000, filter = {}, user_id } = {}) {
```
Comment (Baalmart) [inline src/device-registry/models/KnowYourAirLesson.js:100]: Hi @BenjaminSsempala , isn't this `user_id` already part of the `filter` generated from `utils/generate-filter`?

Comment (BenjaminSsempala) [reply src/device-registry/models/KnowYourAirLesson.js:100]: @Baalmart No, For the Kya Lesson, the `user_id` not part of the filter.

Comment (Baalmart) [reply src/device-registry/models/KnowYourAirLesson.js:100]: Alright @BenjaminSsempala

File: src/device-registry/utils/generate-filter.js [hunk 2]
```diff
@@ -981,7 +981,7 @@ const generateFilter = {
         filter["_id"] = ObjectId(progress_id);
       }
       if (user_id) {
-        filter["user_id"] = ObjectId(user_id);
+        filter["user_id"] = user_id;
```
Comment (Baalmart) [inline src/device-registry/utils/generate-filter.js:984]: Hi @BenjaminSsempala , isn't this the `user_id `being added to the filter object?

**Review by Baalmart — COMMENTED** [review]
thanks @BenjaminSsempala , just added a few comments for your review. 
I plan to approve/merge PR today, after your responses.

Thanks!

File: src/device-registry/utils/generate-filter.js (hunk 2 above)
Comment (BenjaminSsempala) [reply src/device-registry/utils/generate-filter.js:984]: @Baalmart Yes but this filter is for KYA progresses, while the other one is for KYA lessons

Comment (BenjaminSsempala) [reply src/device-registry/utils/generate-filter.js:984]: @Baalmart Yes but this filter is for KYA progresses, while the other one is for KYA lessons

File: src/device-registry/utils/create-know-your-air.js [hunk 3]
```diff
@@ -431,6 +433,99 @@ const createKnowYourAir = {
...
+          success: true,
+          message: "Sync successful",
+          data: syncResponse.data,
+          status: httpStatus.OK,
+        } :
+        syncResponse;
+
+
+    } catch (error) {
+      logger.error("error", error);
```
[the remark this answers is not in the document]
Comment (BenjaminSsempala) [reply src/device-registry/utils/create-know-your-air.js]: Thanks @Baalmart I've made these changes

File: src/device-registry/utils/generate-filter.js (hunk 2 above)
Comment (Baalmart) [reply src/device-registry/utils/generate-filter.js:984]: Cool! Thanks!

**Review by Baalmart — APPROVED** [review]

**Pull request merged**